backend/resolvers/queries/showQuery.js

```diff
@@ -3,43 +3,35 @@ import User from '../../models/user'
 import { UserError } from 'graphql-errors'
 import { ADMIN } from '../../constants'
 
+const isRequestingOwnUser = (req, args) => {
+  return req.auth.username !== undefined &&
+    req.auth.username === args.studentUsername
+}
+
 export function show (_, args, req) {
   return Show.findById(args.id)
 }
 
 export function shows (_, args, req) {
-  // If only order is desired, show all shows ordered
-  if (args.orderBy && !args.studentUsername) {
-    return Show.findAll({
-      order: [[args.orderBy.sort, args.orderBy.direction]]
-    })
+  // Students can only look at their own shows
+  if (args.studentUsername && req.auth.type !== ADMIN && !isRequestingOwnUser(req, args)) {
+    throw new UserError('Permission Denied')
   }
 
+  // If only order is desired, show all shows ordered
+  const order = args.orderBy ? {order: [[args.orderBy.sort, args.orderBy.direction]]} : {}
   // If username given, show all shows the student has submitted to
   if (args.studentUsername) {
-    const isRequestingOwnUser = req.auth.username !== undefined &&
-      req.auth.username === args.studentUsername
-    if (req.auth.type !== ADMIN && !isRequestingOwnUser) {
-      throw new UserError('Permission Denied')
-    }
+    const entryStart = req.auth.type !== ADMIN ? { entryStart: {$lt: new Date()} } : {}
     // Get all the shows the student has been on
     // (including as group creator), through entries
     return User.findById(args.studentUsername).then((student) => {
       return student.getOwnAndGroupEntries()
         // Find all connected shows
         .then((entries) => {
           const showIds = entries.map(entry => entry.showId)
-          // If order doesn't matter just return all the shows
-          if (!args.orderBy) {
-            return Show.findAll({ where: { id: showIds } })
-          } else { // Order shows
-            return Show.findAll({
-              where: {
-                id: showIds
-              },
-              order: [[args.orderBy.sort, args.orderBy.direction]]
-            })
-          }
+          const whereShowIds = { where: { id: showIds } }
+          return Show.findAll(Object.assign({}, whereShowIds, order, entryStart))
```

Comment:
This is mostly a flaw in the original design, but we also need to include shows that are active for students (date.now is between entry start and entry end) otherwise there's no way for a student to add to a show.